app/assets/src/components/AlignmentViz.jsx

```diff
@@ -0,0 +1,276 @@
+    let quality_string = mis_matches1 < mis_matches2 ? quality_string1 : quality_string2
+    let white_space_left = this.repeatStr(' ', left_portion.length)
+    let white_space_right = this.repeatStr(' ', right_portion.length)
+    let ref_seq_display = readInfo.refInfo.join("|")
+    let read_seq_display = [left_portion, aligned_portion, right_portion].join("|")
+    let quality_string_display = [white_space_left, quality_string, white_space_right].join("|")
+
+    // generate quality string
+    return {
+      reversed: reversed,
+      read_part: readPart,
+      ref_display: ref_seq_display,
+      read_display: read_seq_display,
+      quality_display: quality_string_display
+    }
+  }
+
+  repeatStr(c, n) {
+    let output = ''
+    for (let i = 0; i < n; i += 1) {
+      output += c
+    }
+    return output
+  }
+
+  generateQualityString(refString, seqString) {
+    let mis_matches = 0
+    let quality_string = ''
+    for (let i = 0; i < seqString.length; i += 1) {
+      let cr = refString[i]
+      let cs = seqString[i]
+      if (cr === 'N' || cs === 'N' || cr === cs) {
+        quality_string += ' '
+      } else {
+        quality_string += 'X'
+        mis_matches += 1
+      }
+    }
+    return [quality_string, mis_matches]
+  }
+
+  complementSeq(seq) {
+    let output = ''
+    for (let c of seq) {
+      switch (c) {
+        case 'A':
+          output += 'T';
+          break;
+        case 'T':
+          output += 'A';
+          break;
+        case 'C':
+          output += 'G';
+          break;
+        case 'G':
+          output += 'C';
+          break;
+        default:
+          output += c
+      }
+    }
+    return output
+  }
+
+  render() {
+    const nopadding = {padding: '0px'}
+    return (
+      <div style={{border: '1px solid'}}>
+        Read Name: {this.props.name} <br/>
+        Percentage Matched: {this.props.metrics[0]} %, Alignment Length: {this.props.metrics[1]}, # Mismatches: {this.props.metrics[2]}, # Gap Openings: {this.props.metrics[3]} <br/>
+        E-value: {this.props.metrics[8]}, Bit Score: {this.props.metrics[9]} <br/>
+        Reference Alignment Range: {this.props.metrics[6]} - {this.props.metrics[7]} <br/>
+       <div style={{whiteSpace: 'pre', fontFamily: 'monospace', overflow: 'scroll'}}>
+         <table>
+           <tbody>
+             <tr><td style={nopadding}>Reference:  </td><td style={nopadding}>{this.parsedAlignment.ref_display}</td></tr>
+             <tr><td style={nopadding}>Read:       </td><td style={nopadding}>{this.parsedAlignment.read_display}</td></tr>
+             <tr><td style={nopadding}>Mismatches: </td><td style={nopadding}>{this.parsedAlignment.quality_display}</td></tr>
+           </tbody>
+         </table>
+       </div>
+      </div>
+    );
+  }
+}
+
+class AccessionViz extends React.Component {
+  constructor(props) {
+    super(props);
+    this.allReads = props.reads
```

Comment:
is there a particular reason why we're assigning `this.allReads = props.reads` instead of just using `props.reads` directly in constructor() and in this class :D?

Reply:
Ah that makes sense! Thanks for letting me know! 👍👍👍